server_tests/test/clientTests.js

```diff
@@ -27,59 +29,67 @@ describe("RUST API TEST SUITE", function() {
-            testClient.getInfo()
-            .then( Comparison.catchExceptions(done, function(res) {
+        it("Expect success. /osc/info returns correct info", function() {
+            return testClient.getInfo()
+            .then( function onSuccess (res) {
                 Comparison.oscInfoOutput(res);
-                done();
-            }));
+            }, shouldNotGetHere);
         });
     });
 
     // OSC STATE
     describe("Testing /osc/state endpoint", function() {
         var sessionId;
 
-        before( function(done) {
-            testClient.startSession()
-            .then( Comparison.catchExceptions(done, function(res) {
+        before( function() {
+            return testClient.startSession()
+            .then( function onSuccess (res) {
                 sessionId = res.body.results.sessionId;
                 Comparison.oscSessionOpOutput(res, {'sessionId': sessionId});
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        it("Expect success. /osc/state endpoint successfully returns state when state has not changed", function(done) {
-            testClient.getState()
-            .then( Comparison.catchExceptions(done, function(res) {
+        it("Expect success. /osc/state endpoint successfully returns state when state has not changed", function() {
+            return testClient.getState()
+            .then( function onSuccess (res) {
                 Comparison.oscStateOutput(res, {'sessionId' : sessionId});
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        it("Expect success. confirming /osc/state endpoint returns correct value when state has changed", function(done) {
+        it("Expect success. confirming /osc/state endpoint returns correct value when state has changed", function() {
             var oldFingerprint;
 
-            testClient.getState()
-            .then( Comparison.catchExceptions(done, function(res) {
+            return testClient.getState()
+            .then( function onSuccess (res) {
                 Comparison.oscStateOutput(res, {'sessionId' : sessionId});
-                oldFingerprint = res.fingerprint;
+                oldFingerprint = res.body.fingerprint;
                 return testClient.closeSession(sessionId);
-            }))
-            .then( Comparison.catchExceptions(done, function(res) {
+            }, shouldNotGetHere)
+            .then( function onSuccess (res) {
                 Comparison.oscCloseSessionOutput(res);
                 return testClient.getState();
-            }))
-            .then( Comparison.catchExceptions(done, function(res) {
+            }, shouldNotGetHere)
+            .then( function onSuccess (res) {
                 Comparison.oscStateOutput(res, {'sessionId': '', 'fingerprint': oldFingerprint});
-                done();
-            }));
+            }, shouldNotGetHere);
```

Comment:
i think the .catch(shouldNotGetHere) everywhere may make sense. It appears to me that the current configuration is pretty darn close to entering the callback hell?